cli/js/web/url.ts

```diff
@@ -259,9 +266,7 @@ export class URLImpl implements URL {
         value = `#${value}`;
       }
       // hashes can contain % and # unescaped
-      parts.get(this)!.hash = escape(value)
-        .replace(/%25/g, "%")
-        .replace(/%23/g, "#");
+      parts.get(this)!.hash = encoding.encodeHash(value);
```

Comment:
Since you moved the responsability of this section to a method, this comment would be most valuable as a docstring there in my opinion.